aws-janitor/resources/internet_gateways.go

```diff
@@ -77,36 +74,46 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
-			var publicIPsToRelease []*string
+			if opts.DisassociatePublicIP {
+				var publicIPsToRelease []*string
 
-			pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
-				for _, eni := range page.NetworkInterfaces {
-					publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
+				pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+					for _, eni := range page.NetworkInterfaces {
+						publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
+					}
+					return true
 				}
-				return true
-			}
 
-			if err := svc.DescribeNetworkInterfacesPages(&ec2.DescribeNetworkInterfacesInput{
-				Filters: []*ec2.Filter{
-					{
-						Name:   aws.String("vpc-id"),
-						Values: []*string{aws.String(*att.VpcId)},
+				if err := svc.DescribeNetworkInterfacesPages(&ec2.DescribeNetworkInterfacesInput{
+					Filters: []*ec2.Filter{
+						{
+							Name:   aws.String("vpc-id"),
+							Values: []*string{aws.String(*att.VpcId)},
+						},
 					},
-				},
-			}, pageFunc); err != nil {
-				return err
-			}
+				}, pageFunc); err != nil {
+					logger.Warningf("fail to get public IP for vpc %s: %v", *att.VpcId, err)
+				}
```

Comment:
We log the error but not return, the reason is that some times, the DescribeNetworkInterfacesPages returns not-nil error, making the rest resources clean up logic terminates in the middle.